e3db/identity.py

```diff
@@ -6,17 +6,94 @@
     from .nist_crypto import NistCrypto as Crypto
 else:
     from .sodium_crypto import SodiumCrypto as Crypto
+from typing import Tuple
+from e3db.tsv1_auth import E3DBTSV1Auth
+import requests
+import json
+from .exceptions import APIError, LookupError, CryptoError, QueryError, ConflictError, NoteValidationError, UnsupportedAPIResponse
 
+TOZID_LOGIN_HEADER = "X-TOZID-LOGIN-TOKEN"
 
 class Identity:
     """
     This is a placeholder class pending further development of Identity Login
-    TODO:  Contains a realm, have a client, have all of its OAuth tokens
+    TODO WASH-902:  Contains a realm, a storage client, have all of its OAuth tokens
     """
+    DEFAULT_API_URL = "https://api.e3db.com"
 
+    def __init__(self):
+        """
+        TODO
+        """
 
     @staticmethod
-    def derive_note_creds(user_name: str, password: str, realm_name: str):
+    def identity_login(user_name: str, password: str, realm_name, app_name: str, api_url: str=DEFAULT_API_URL) -> dict:
+        """
+        A factory mehod to login an existing user, get the stored identity credentials for a user,
+        create a client for them, and return an Identity object.
+
+        TODO: Change param from realm_name & app_name to a Realm Object
+        TODO WASH-902: Currently returns a config object - need to create an Id_Client class to create with the config
+
+        The username and password are used to derive encryption keys used to fetch a pre-stored
+        note which contains the users identity credentials.
+
+        Only password based realm types are supported at this time.
+        Brokered Login is not supported at this time.
+        MFA is not supported
+
+        Parameters
+        ----------
+        user_name : str
+
+        password : str
+
+        realm_name : str
+            will be converted to lower case
+        
+        app_name : str
+
+        api_url: str
+            Defaults to https://api.e3db.com
+
+        Retruns
+        -------
+        dict
+            config object for creating an identity client
+        """
+        realm_name = realm_name.lower()
+        note_name, key_pair, signing_key_pair = Identity.derive_note_creds(user_name, password, realm_name)
```

Comment:
Can we fetch the realm info, so we can add realm domain

Reply:
great catch

Reply:
This has been updated